pretty_midi/pretty_midi.py

```diff
@@ -1353,6 +1353,37 @@ def adjust_meta(events):
         # Update the tick-to-time mapping
         self._update_tick_to_time(self._tick_scales[-1][0] + 1)
 
+    def crop(self, start_time=0.0, end_time=None):
+        """Crops the MIDI object, keeping only the segment from ``start_time`` to ``end_time``.
+
+        Parameters
+        ----------
+        start_time : float
+            Segment start time, in seconds. Defaults to 0.0.
+        end_time : float
+            Segment end time, in seconds. Defaults to self.end_time().
+
+        """
+        midi_end_time = self.get_end_time()
+
+        # Enforce that start_time is non-negative and end_time is lower than the MIDI object's end time. 
+        if start_time < 0.0:
+            warnings.warn('start_time is less than 0; automatically adjusting to 0.')
```

Comment:
This should be a ValueError, and let the user fix it.

Reply:
Done.